packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -125,70 +128,109 @@ contract BridgePool is Testable {
-        });
-        bytes memory priceRequestAncillaryData = getRelayAncillaryData(newDeposit.relayData);
+        newRelay.relayState = RelayState.PendingSlow;
+        newRelay.relayType = RelayType.Slow;
 
         // Request a price for the relay identifier and propose "true" optimistically. These methods will pull the
         // (proposer reward + proposer bond + final fee) from the caller.
-        // Note: We can't simply set the price request timestamp equal to the `depositTimestamp`, which is dependent
-        // on the L2 VM on which the DepositContract is deployed. Imagine if the timestamps on the L2 have an offset
-        // that are always "in the future" relative to L1 blocks, then the OptimisticOracle would always reject
-        // requests.
-        uint256 requestTimestamp = getCurrentTime();
-        _requestOraclePriceRelay(l1Token, amount, requestTimestamp, priceRequestAncillaryData, proposerRewardPct);
-        _proposeOraclePriceRelay(l1Token, amount, depositTimestamp, priceRequestAncillaryData);
-
-        // TODO: There is more data we'd like to emit, such as depositId, but we are limited by how many variables
-        // we can fit into an event. Perhaps we need multiple events? Or think more critically about what information
-        // clients will want.
+        // Note: We need to set a new price request timestamp instead of default setting to equal to the
+        // `depositTimestamp`, which is dependent on the L2 VM on which the DepositContract is deployed. Imagine if
+        // the timestamps on the L2 have an offset that are always "in the future" relative to L1 blocks, then the
+        // OptimisticOracle would always reject requests.
+        _requestOraclePriceRelay(
+            l1Token,
+            amount,
+            getCurrentTime(),
+            getRelayAncillaryData(relayData),
+            proposerRewardPct
+        );
+        _proposeOraclePriceRelay(l1Token, amount, depositTimestamp, getRelayAncillaryData(relayData));
+
+        // Since we only store the hash of the relay data contents, its important that we emit all relay params in an
+        // event. This should aid off-chain clients who want to modify/query the deposit.
         emit DepositRelayed(
+            depositId,
             l2Sender,
             depositTimestamp,
             recipient,
             l1Token,
-            msg.sender,
             amount,
+            msg.sender,
+            maxFeePct,
             proposerRewardPct,
             realizedFeePct,
-            bridgeAdmin.depositContract()
+            keccak256(getRelayAncillaryData(relayData))
         );
     }
 
-    function speedUpRelay(uint256 depositId) public {}
+    function speedUpRelay(RelayDataContents memory _relayData) public {
+        // Alternatively, force client to hash off-chain and pass `bytes32 _relayAncillaryDataHash` as param.
+        Relay storage relay = relays[keccak256(getRelayAncillaryData(_relayData))];
```

Comment:
Seems a little wasteful to hash the entire ancillary data each time since it means we have to construct the ancillary data on each call (lots of string concats). Any reason we can't just hash the values as we were before and only construct the ancillary data when calling to the OO? Or are there disadvantages to doing that?